test-suite/src/test/scala/org/scalajs/testsuite/javalib/TreeSetTest.scala

```diff
@@ -0,0 +1,332 @@
+      val ts = new TreeSet[Double]()
+
+      expect(ts.add(11111.0)).toBeTruthy
+      expect(ts.size()).toEqual(1)
+      expect(ts.contains(11111.0)).toBeTruthy
+      expect(ts.iterator.next()).toEqual(11111.0)
+
+      expect(ts.add(Double.NaN)).toBeTruthy
+      expect(ts.size()).toEqual(2)
+      expect(ts.contains(Double.NaN)).toBeTruthy
+      expect(ts.contains(+0.0)).toBeFalsy
+      expect(ts.contains(-0.0)).toBeFalsy
+
+      expect(ts.remove(Double.NaN)).toBeTruthy
+      expect(ts.add(+0.0)).toBeTruthy
+      expect(ts.size()).toEqual(2)
+      expect(ts.contains(Double.NaN)).toBeFalsy
+      expect(ts.contains(+0.0)).toBeTruthy
+      expect(ts.contains(-0.0)).toBeFalsy
+
+      expect(ts.remove(+0.0)).toBeTruthy
+      expect(ts.add(-0.0)).toBeTruthy
+      expect(ts.size()).toEqual(2)
+      expect(ts.contains(Double.NaN)).toBeFalsy
+      expect(ts.contains(+0.0)).toBeFalsy
+      expect(ts.contains(-0.0)).toBeTruthy
+
+      expect(ts.add(+0.0)).toBeTruthy
+      expect(ts.add(Double.NaN)).toBeTruthy
+      expect(ts.contains(Double.NaN)).toBeTruthy
+      expect(ts.contains(+0.0)).toBeTruthy
+      expect(ts.contains(-0.0)).toBeTruthy
+    }
+
+    when("compliant-asinstanceofs").
+    it("should throw exceptions on access outside bound on views") {
+      val l = asJavaCollection(Set(2, 3, 6))
+      val ts = new TreeSet[Int](l)
+
+      val hs1 = ts.headSet(5, true)
+      expect(hs1.add(4)).toBeTruthy
+      expect(hs1.add(5)).toBeTruthy
+      expectThrows[IllegalArgumentException](hs1.add(6))
+
+      ts.clear()
+      ts.addAll(l)
+
+      val hs2 = ts.headSet(5, false)
+      expect(hs2.add(4)).toBeTruthy
+      expectThrows[IllegalArgumentException](hs2.add(5))
+
+      ts.clear()
+      ts.addAll(l)
+
+      val ts1 = ts.tailSet(1, true)
+      expect(ts1.add(7)).toBeTruthy
+      expect(ts1.add(1)).toBeTruthy
+      expectThrows[IllegalArgumentException](ts1.add(0))
+
+      ts.clear()
+      ts.addAll(l)
+
+      val ts2 = ts.tailSet(1, false)
+      expect(ts2.add(7)).toBeTruthy
+      expectThrows[IllegalArgumentException](ts2.add(1))
+
+      ts.clear()
+      ts.addAll(l)
+
+      val ss1 = ts.subSet(1, true, 5, true)
+      expect(ss1.add(4)).toBeTruthy
+      expect(ss1.add(1)).toBeTruthy
+      expectThrows[IllegalArgumentException](ss1.add(0))
+      expect(ss1.add(5)).toBeTruthy
+      expectThrows[IllegalArgumentException](ss1.add(6))
+
+      ts.clear()
+      ts.addAll(l)
+
+      val ss2 = ts.subSet(1, false, 5, false)
+      expect(ss2.add(4)).toBeTruthy
+      expectThrows[IllegalArgumentException](ss2.add(1))
+      expectThrows[IllegalArgumentException](ss2.add(5))
+    }
+
+    it("should throws exception in case of null elements and default ordering") {
+      val hs = new TreeSet[String]()
+
+      expect(hs.add("ONE")).toBeTruthy
+      expect(hs.contains("ONE")).toBeTruthy
+      expect(hs.contains("TWO")).toBeFalsy
+      expectThrows[NullPointerException](hs.add(null))
```

Comment:
This will be pass if `treeSetTestFactory.allowsNullElement` is `false`, test also the other case.